### PR TITLE
py-exif: update to 2.1.2, major portfile rewrite

### PR DESCRIPTION
**Pull request by chrstphrchvz**
- Whitespace changes
 - Use github portgroup; distfile no longer hosted on SourceForge
 - Remove freebsd from platforms
 - Add build dependency pyXX-setuptools; remove unneeded setup.py script
 - Add subport for Python 3.7

#### Description

The decision to keep the port was made along with suggested improvements: https://github.com/macports/macports-ports/pull/3374#issuecomment-452427717

The portfile is updated to use the 2.1.2 release from GitHub. I think it would take me a few more days to find a more recent commit or create patches for beneficial changes since the 2.1.2 release. So far I have tested the 2.1.2 release and the latest commit (https://github.com/ianare/exif-py/commit/ea5af101ff0a4c478cc05b1a193372d4b33218d8), and each had different minor bugs present. The project has a repository of test images that are useful for finding if any bugs are fixed or introduced: https://github.com/ianare/exif-samples/

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.2 18C54
Xcode 10.1 10B61
Python 3.7.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] ~~referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?~~

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-exif/Portfile [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-distname		EXIFpy_${version}
-checksums		md5 7df9f94e98b6ff5da872105a11069356 \
-			sha1 859411ec882fc297a4aa4f1368aa5c2a14688573 \
-			rmd160 26bbc5fbdd01b7c8cf472895f66a9f633b0510fd
+checksums           rmd160  c24bf810a72a7a176833b7e1d564349d316fe868 \
+                    sha256  53b0fa8e33071320ab1d8ae4713a7bea859ec166bc3774df9003b25ffea94b90 \
+                    size    32200
 
-python.versions	27
+python.versions     27 33 34 35 36 37
```
Comment (mf2k) [inline python/py-exif/Portfile]: Why add all of the old python 3 versions? Please just add py37.

Comment (chrstphrchvz) [reply python/py-exif/Portfile]: I was not aware older versions should not be added to portfiles. I will remove the older ones, however the project only acknowledges it has been tested on versions up to 3.5.

**Comment by chrstphrchvz** [comment]
Pushed new commit e51cc90 so that the only Python 3 subport being added is for 3.7.

File: python/py-exif/Portfile (hunk 1 above)
Comment (mf2k) [reply python/py-exif/Portfile]: We are actively removing old python sub-ports. Only the latest stable version should be added to existing ports. Of course if it does not work on python >3.5, only py35 should be added.

**Comment by pmetzger** [comment]
@mf2k How do you feel about the new version? (Feel free to merge it if you're happy.)

**Comment by mf2k** [comment]
Someone needs to test it with python 3.7. It sounds like upstream only has tested with python 3.5.

**Comment by mojca** [comment]
Yes, someone needs to test with 3.7 in any case.

But unless it's known to be incompatible with 3.7 (not tested with 3.7 doesn't count as incompatible), we should not be adding 3.5 (as the only supported version).

**Comment by chrstphrchvz** [comment]
I have been testing this exclusively on 3.7 and haven't observed any issues that would be due to 3.6/3.7. Recent Linux distributions with only 3.6/3.7 are packaging this module seemingly as-is. I have gone ahead and opened an issue with the project regarding Python 3.6+ support: https://github.com/ianare/exif-py/issues/88

**Comment by pmetzger** [comment]
Merged. Thanks, @chrstphrchvz!

**Pull request merged**